chromium_src/chrome/browser/sync/sync_service_factory.cc

```diff
@@ -3,10 +3,35 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "chrome/browser/sync/sync_service_factory.h"
+
+#include "base/command_line.h"
 #include "brave/browser/sync/brave_sync_service_impl_delegate.h"
+#include "brave/components/brave_sync/brave_sync_prefs.h"
 #include "brave/components/sync/service/brave_sync_service_impl.h"
 #include "chrome/browser/history/history_service_factory.h"
 #include "chrome/browser/sync/device_info_sync_service_factory.h"
+#include "components/sync/base/command_line_switches.h"
+
+// Below includes are just to prevent redefining of
```

Comment:
These included seems nasty, 
@goodov Do we have any known alternatives to avoid this, other than direct patching?

Reply:
yeah, this is too messy.

I think we should stop using `syncer::kSyncServiceURL` for setting the URL via the command line, but instead do this:
1. override `syncer::internal::kSyncServerUrl` and `syncer::internal::kSyncDevServerUrl` to be `BUILDFLAG(BRAVE_SYNC_ENDPOINT)`.
2. modify `BraveGetSyncServiceURL` to also use the pref value if `syncer::kSyncServiceURL` flag is not passed.
3. remove `syncer::kSyncServiceURL` setting in `//brave/app/brave_main_delegate.cc`.
4. support `AdjustSyncServiceUrlForAndroid` in `BraveGetSyncServiceURL`.

Reply:
cc @AlexeyBarabash ^ does this sound correct? aren't there any hidden shenanigans on iOS/Android if we go this way?

Reply:
Sorry, @jagadeshjai  @goodov  I have conmpletly missed this PR and the question.

all above sounds correct and I see the code is now much cleaner. 

No hidden  shenanigans on Android.

iOS should also be ok. Just in case lets ask @Brandon-T, does iOS have any code to setup custom sync server url?

I just have tested the PR on Android and Desktop - works fine.